cpp/src/arrow/ipc/reader.h

```diff
@@ -301,9 +317,39 @@ class ARROW_EXPORT CollectListener : public Listener {
     return record_batches_;
   }
 
+  /// \return the all decoded metadatas
+  std::vector<std::shared_ptr<KeyValueMetadata>> metadatas() const { return metadatas_; }
+
+  /// \return the number of collected record batches
+  size_t num_record_batches() const { return record_batches_.size(); }
```

Comment:
Return `int64_t` here?

Reply:
Ah, yes. Other places use `int64_t` or `int` for the number of record batches.